cinder/volume/manager.py

```diff
@@ -680,6 +680,43 @@ def _set_resource_host(self, resource):
             resource.host = volume_utils.append_host(self.host, pool)
             resource.save()
 
+    def _propogate_volume_scheduler_hints(self, context, volume):
+        """Ensure metadata hints are propagated to referenced volumes."""
+        hints = volume_utils.get_scheduler_hints_from_volume(volume)
+        if hints:
+            LOG.debug("Found hints for %(volume)s - %(hints)s",
+                      {'volume': volume.id, 'hints': hints})
+            for hint_key in hints.keys():
+                for vol_id in hints[hint_key]:
+                    try:
+                        meta_vol = objects.Volume.get_by_id(context, vol_id)
+                        LOG.debug("Looking for hints in %s", meta_vol.id)
+                        # Because pep8 error otherwise with lines 696 and 710
+                        vut = volume_utils
+                        subhints = vut.get_scheduler_hints_from_volume(
+                            meta_vol
+                        )
+                        LOG.debug("hints? %s", subhints)
+                        if subhints:
+                            if (hint_key in subhints and
+                                    volume.id not in subhints[hint_key]):
+                                subhints[hint_key].append(volume.id)
+                            else:
+                                subhints[hint_key] = [volume.id]
+                        else:
+                            subhints = {hint_key: [volume.id]}
+
+                        LOG.debug("Set hints in metadata %s", subhints)
+                        md = vut.set_scheduler_hints_to_volume_metadata(
+                            subhints, meta_vol.metadata
+                        )
+                        LOG.debug("metadata  %s", md)
+                        meta_vol.metadata = md
+                        meta_vol.save()
+                    except Exception as ex:
+                        LOG.error("Failed to set scheduler hints")
+                        LOG.error(ex)
```

Comment:
This could be a single `LOG.exception("Failed to set scheduler hints")`, I guess. Any reason for not printing the stacktrace and just logging the exception message i.e. the stringified exception - in another line?